application/actions/messages.go

```diff
@@ -0,0 +1,28 @@
+package actions
+
+import (
+	"errors"
+
+	"github.com/silinternational/wecarry-api/api"
+	"github.com/silinternational/wecarry-api/models"
+)
+
+func convertMessagesToAPIType(messages models.Messages) (api.Messages, error) {
+	var output api.Messages
+	if err := api.ConvertToOtherType(messages, &output); err != nil {
+		err = errors.New("error converting messages to apitype.messages: " + err.Error())
+		return nil, err
+	}
+
+	// Hydrate the thread's messages
+	for i := range messages {
```

Comment:
I think this would be a little more clear to me if we iterated through `output` and then updated the current index `SentBy`. It may be safe to assume same ordering between original `messages` and `output`, but it'd be easier to trust and read if it was updating the thing it was iterating through.

Reply:
:+1: